src/net_processing.cpp

```diff
@@ -5217,7 +5218,7 @@ void PeerManagerImpl::EvictExtraOutboundPeers(std::chrono::seconds now)
             // If this is the only connection on a particular network that is
             // OUTBOUND_FULL_RELAY or MANUAL, protect it.
             if (!m_connman.MultipleManualOrFullOutboundConns(pnode->addr.GetNetwork())) return;
-            if (state->m_last_block_announcement < oldest_block_announcement || (state->m_last_block_announcement == oldest_block_announcement && pnode->GetId() > worst_peer)) {
+            if (!oldest_block_announcement.has_value() || (state->m_last_block_announcement == *oldest_block_announcement && pnode->GetId() > worst_peer)) {
```

Comment:
This looks wrong to me, i think it should be:

```suggestion
            if (state->m_last_block_announcement < oldest_block_announcement.value_or(NodeSeconds::max()) || (state->m_last_block_announcement == *oldest_block_announcement && pnode->GetId() > worst_peer)) {
```

Reply:
Thanks, good catch! I force-pushed to fix this in a slightly different way that seems more clear to me.